lib/authz/permissions_test.go

```diff
@@ -451,3 +444,90 @@ type fakeCtxChecker struct {
 func (c *fakeCtxChecker) GetAccessState(_ types.AuthPreference) services.AccessState {
 	return c.state
 }
+
+type testClient struct {
+	services.ClusterConfiguration
+	services.Trust
+	services.Access
+	services.Identity
+	types.Events
+}
+
+func newTestResources(t *testing.T) (*testClient, *services.LockWatcher, Authorizer) {
+	backend, err := memory.New(memory.Config{})
+	require.NoError(t, err)
+
+	clusterConfig, err := local.NewClusterConfigurationService(backend)
+	require.NoError(t, err)
+	caSvc := local.NewCAService(backend)
+	accessSvc := local.NewAccessService(backend)
+	identitySvc := local.NewIdentityService(backend)
+	eventsSvc := local.NewEventsService(backend)
+
+	testClient := &testClient{
+		ClusterConfiguration: clusterConfig,
+		Trust:                caSvc,
+		Access:               accessSvc,
+		Identity:             identitySvc,
+		Events:               eventsSvc,
```

Comment:
Do any of these need to be closed? What about `backend`? `lockWatcher`?

WDYT about adding a Close function to testClient, or returning a `func() error` the caller can defer-close?

Reply:
It looks like only the backend needs to be closed, so I'll add this to `t.Cleanup` which should cover this.